### PR TITLE
Claim & Restake Modal

### PR DESCRIPTION
**Pull request by swindesr**
Ref: https://github.com/multisig-labs/app.gogopool.com/issues/442

1. New Pending/Error/Success screens for the flow
2. New component structure for multi-step modal
3. `SimpleSteps` component to display the current progress through multi-step modal
4. Some cleanup in the dashboard, pass fewer props
5. New layout/visual update for the modal to make new stake modal/unstake modals per the figma mocks

image

Future work remaining:
1. Swap "Success" screen icon to the one shown in figma
2. Swap error screen to updated version with detailed errors

### CODE REVIEW

File: src/hooks/useClaimNodeOp.ts [hunk 1]
```diff
@@ -50,7 +54,8 @@ export const useClaimAndRestake = (claimAmount: BigNumber, restakeAmount: BigNum
 
   return useContractWrite({
     ...config,
-    onSuccess(data) {
+    onSuccess: (data) => {
+      onConfirmTransaction()
```
Comment (swindesr) [inline src/hooks/useClaimNodeOp.ts:58]: Added callback here to support acknowledging when the user has `Confirm`'d the transaction from their wallet. For the modal, we use this to kick us over to step#3 which shows success/loading/error screens

File: src/modules/components/Modal/ClaimAndRestake/ClaimAndRestakeModal.tsx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+import { Title } from '@/common/components/Card'
+import { useClaimAndRestake } from '@/hooks/useClaimNodeOp'
+import { useGetFutureRatio } from '@/hooks/useGetFutureRatio'
+import { useGetContractCollateralizationRatio, useGetGGPRewards } from '@/hooks/useStake'
+
+interface ClaimAndRestakeModalProps {
+  onClose(): void
+}
+
+export const ClaimAndRestakeModal: FunctionComponent<ClaimAndRestakeModalProps> = ({ onClose }) => {
```
Comment (swindesr) [inline src/modules/components/Modal/ClaimAndRestake/ClaimAndRestakeModal.tsx:35]: Fairly similar to old modal structure, but some key differences:

1. gave more control the the `ClaimAndRestakeModal` parent component. It handles the state of the modal and chooses which body/footer segments to display based on the state.
2. pushed server state lookups into child components where possible.
3. Add support for new styling and form step indicator

File: src/modules/components/Modal/ClaimAndRestake/ClaimRestakeStepOne.tsx [hunk 3]
```diff
@@ -1,81 +1,113 @@
...
-          {restakeAmount.eq(0) ? 'Claim All...' : 'Claim and Restake...'}
-        </Button>
-      </Flex>
-    </div>
+      <Box pb="24" pt="16" px="20">
+        <Title fontWeight="500" lineHeight="40px" width="300px">
+          How much do you want to restake?
+        </Title>
+        <Box pt="5">
+          <HStack
```
Comment (swindesr) [inline src/modules/components/Modal/ClaimAndRestake/ClaimRestakeStepOne.tsx:55]: maybe worth keeping in mind that we are duplicating this across a few modals now. It's in about 50% of the `BigNumberInput` usages so could be worth extracting to a common wrapper

**Review by jacktrusler — COMMENTED** [review]
All sensible changes, I might have missed something but looks solid to me. New modal looks great. 

I made a branch off yours where i extracted a function and made a wrapper for BNinputs on the modals. You can choose to merge in what you like, then i'll approve.

**Pull request merged**